sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -576,6 +576,11 @@ class Dataset[T] private[sql](
     val parsedDelay =
       Option(CalendarInterval.fromString("interval " + delayThreshold))
         .getOrElse(throw new AnalysisException(s"Unable to parse time delay '$delayThreshold'"))
+    val delayMs = {
+      val millisPerMonth = CalendarInterval.MICROS_PER_DAY / 1000 * 31
+      parsedDelay.milliseconds + parsedDelay.months * millisPerMonth
+    }
+    assert(delayMs >= 0, s"delay threshold should not be a negative time: $delayThreshold")
```

Comment:
Why compute all this -- don't you just mean to assert about `delayThreshold`? this derived value can only be negative if the input is right?